nexus/tests/integration_tests/disks.rs

```diff
@@ -830,6 +793,317 @@ async fn test_disk_reject_total_size_not_divisible_by_min_disk_size(
     );
 }
 
+// Test disks backed by multiple region sets (one region set being three regions)
+#[nexus_test]
+async fn test_disk_backed_by_multiple_region_sets(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let client = &cptestctx.external_client;
+
+    // Create three zpools, all 10 gibibytes, each with one dataset
+    let mut test = DiskTest::new(&cptestctx).await;
+
+    // Create another three zpools, all 10 gibibytes, each with one dataset
+    test.add_zpool_with_dataset(10).await;
+    test.add_zpool_with_dataset(10).await;
+    test.add_zpool_with_dataset(10).await;
+
+    create_org_and_project(client).await;
+
+    // Ask for a 20 gibibyte disk.
+    let disk_size = ByteCount::try_from(20u64 * 1024 * 1024 * 1024).unwrap();
+    let disks_url = get_disks_url();
+    let new_disk = params::DiskCreate {
+        identity: IdentityMetadataCreateParams {
+            name: DISK_NAME.parse().unwrap(),
+            description: String::from("sells rainsticks"),
+        },
+        disk_source: params::DiskSource::Blank {
+            block_size: params::BlockSize::try_from(512).unwrap(),
+        },
+        size: disk_size,
+    };
+
+    NexusRequest::new(
+        RequestBuilder::new(client, Method::POST, &disks_url)
+            .body(Some(&new_disk))
+            // TODO: this fails! the current allocation algorithm does not split
```

Comment:
Nit: can we attach a bug?

Reply:
Opened https://github.com/oxidecomputer/omicron/issues/1644